src/nvim/terminal.c

```diff
@@ -315,8 +315,8 @@ static int on_osc(int command, VTermStringFragment frag, void *user)
   }
   kv_concat_len(term->termrequest_buffer, frag.str, frag.len);
   if (frag.final) {
-    schedule_termrequest(user, term->termrequest_buffer.items, term->termrequest_buffer.size);
-    kv_init(term->termrequest_buffer);
+    char *payload = xmemdupz(term->termrequest_buffer.items, term->termrequest_buffer.size);
```

Comment:
This is my mistake, I asked you to use `xmemdupz` but since we are passing the length to `schedule_termrequest` we don't need the null sentinel, so `xmemdup` will do. I'll fix that up.